src/main/scala/viper/silver/reporter/Message.scala

```diff
@@ -210,6 +211,30 @@ case class InvalidArgumentsReport(tool_signature: String, errors: List[AbstractE
   override val name: String = s"invalid_args_report"
 }
 
+object BackendSubProcessStages extends Enumeration {
+  type BackendSubProcessStage = Value
+  val BeforeInputSent         = Value(1, "before_input_sent")
+  val AfterInputSent          = Value(2, "after_input_sent")
+  val onOutput                = Value(3, "on_output")
+  val OnError                 = Value(4, "on_error")
+  val BeforeTermination       = Value(5, "before_termination")
+  val OnExit                  = Value(6, "on_exit")
+  val AfterTermination        = Value(7, "after_termination")
+}
+
+case class BackendSubProcessReport(tool_signature: String, process_exe: String,
+                                   phase: BackendSubProcessStage, pid_maybe: Option[Long] = None) extends Message {
+
+  override def toString: String =
```

Comment:
If you expect the creation of very many instances of `BackendSubProcessReport`, a `lazy val toString` would be better than a `def`